CONTRIBUTING.md

```diff
@@ -0,0 +1,481 @@
+# Solana Developer Content Contributing Guide
+
+We love your input! We want to make contributing to the
+[Solana Developer](https://solana.com/developers) content as easy and
+transparent as possible, whether it's:
+
+- Reporting a bug
+- Submitting a fix
+- Discussing the current state of this repo
+- Proposing new features
+
+## Key points
+
+- called the "developer content" repo or "developer content api"
+- written in markdown with yaml frontmatter
+- used yaml frontmatter, with structure enforced with Contentlayer
+- all pages are grouped into "content types"
+- supports some select custom components
+- publicly displayed via the UI of [solana.com](https://solana.com) (located in
+  a different repo)
+- content translations are supported via Crowdin
+
+## Content guidelines
+
+Since this content within this repo is meant to be publicly displayed on
+`solana.com`, we have very specific guidelines for the content that can be
+merged into this repo (and eventually displayed on solana.com). If you would
+like to submit changes to existing content or add new content, all of these
+guidelines should be observed:
+
+### Avoid official recommendations
+
+Avoid any language around making "official recommendations" such as "I recommend
+product X" or "product Y is the best". The content within this repo will be
+publicly published on solana.com which is maintained by the
+[Solana Foundation](https://solana.org). As such, any "product recommendations"
+may appear as if coming directly the Solana Foundation. The Solana Foundation
+does not make official recommendations for products, but rather help share what
+options are available in the broader Solana ecosystem.
+
+### Avoid "picking favorites"
+
+Avoid language similar to "service X is my favorite". When talking about or
+naming specific products within the Solana ecosystem, writers and maintainers
+should make their best attempt to list multiple options that meet similar use
+cases. As a general rule of thumb, try to share at least 3-4 options of similar
+products or services, not just a single named one. For example, when talking
+about wallet providers, a writer could list Phantom, Solflare, Backpack, and
+Ultimate. When talking about RPC providers, a writer could list Triton, Helius,
```

Comment:
For RPC providers, shall just say reference the RPC list at https://solana.com/rpc? Then no one has to name a specific one and say just grab a RPC from the page.